engine/engine.go

```diff
@@ -129,6 +133,19 @@ func New(opts Opts) *compatibilityEngine {
 		level.Debug(opts.Logger).Log("msg", "lookback delta is zero, setting to default value", "value", 5*time.Minute)
 	}
 
+	metrics := &engineMetrics{
+		currentQueries: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "thanos",
+			Subsystem: "engine",
+			Name:      "queries",
+			Help:      "The current number of queries being executed or waiting.",
+		}),
+	}
+
+	if opts.EngineOpts.Reg != nil {
+		opts.EngineOpts.Reg.MustRegister(metrics.currentQueries)
```

Comment:
Let's use `promauto.With(opts.Reg)`? It seems like `opts.Reg` is a mandatory parameter.